src/Compilers/Core/Portable/Diagnostic/DiagnosticSeverity.cs

```diff
@@ -27,7 +27,12 @@ public enum DiagnosticSeverity
         /// <summary>
         /// Something not allowed by the rules of the language or other authority.
         /// </summary>
-        Error = 3
+        Error = 3,
+
+        /// <summary>
+        /// Something that can be potentially improved.
+        /// </summary>
+        Suggestion = 4,
```

Comment:
Dustin is correct with these parts:

> I find a design where Suggestion is a custom tag for Info to be enormously fragile. With that, we would unnecessarily create invalid combinations of severities and tags since "Suggestion" is nonsensical when combined with "Error" and "Warning". If Suggestion is upgraded to a new severity, the combinations of tags and severities are all sensible.

> FWIW, the notion that UI presentation is controlled by tags is simply not reality today. We have one tag that controls UI presentation, but the bulk of UI presentation is handled by severities (e.g. error and warning squiggles).

Suggestion should not be a custom tag, because it doesn't make sense at all with all the severities we have today.  What would it mean to be "Error + Suggestion"?  or "Hidden + Suggestion".

The reason we have "Unnecessary" as a tag is because it augments the existing severities.  i.e. it is completely reasonable to have:

Hidden+Unnecessary
Info+Unnecessary
Warning+Unnecessary
Error+Unecessary

It even makes sense to have "Suggestion+Unnecessary"

"Unnecessary" is orthogonal to severity.

On hte other hand, Suggestion is not orthogonal.  Something cannot be both a Suggestion and an Error.  It has no meaning.